WalletWasabi.Fluent/ViewModels/Wallets/Advanced/WalletCoins/WalletCoinsViewModel.cs

```diff
@@ -23,35 +22,100 @@ namespace WalletWasabi.Fluent.ViewModels.Wallets.Advanced.WalletCoins;
 [NavigationMetaData(Title = "Wallet Coins (UTXOs)")]
 public partial class WalletCoinsViewModel : RoutableViewModel
 {
-	private readonly WalletViewModel _walletViewModel;
 	private readonly IObservable<Unit> _balanceChanged;
-	private readonly ObservableCollectionExtended<WalletCoinViewModel> _coins;
-	private readonly SourceList<WalletCoinViewModel> _coinsSourceList = new();
-	[AutoNotify] private FlatTreeDataGridSource<WalletCoinViewModel>? _source;
+	private readonly WalletViewModel _walletViewModel;
+	private ObservableAsPropertyHelper<bool>? _isAnySelected;
+	[AutoNotify] private FlatTreeDataGridSource<WalletCoinViewModel> _source;
 
 	public WalletCoinsViewModel(WalletViewModel walletViewModel, IObservable<Unit> balanceChanged)
 	{
-		SetupCancel(false, true, true);
-		NextCommand = CancelCommand;
 		_walletViewModel = walletViewModel;
 		_balanceChanged = balanceChanged;
-		_coins = new ObservableCollectionExtended<WalletCoinViewModel>();
+		SetupCancel(false, true, true);
+		NextCommand = CancelCommand;
 
 		SkipCommand = ReactiveCommand.CreateFromTask(OnSendCoins);
+	}
+
+	public bool IsAnySelected => _isAnySelected?.Value ?? false;
 
-		AnySelected = _coinsSourceList
+	protected override void OnNavigatedTo(bool isInHistory, CompositeDisposable disposables)
+	{
+		var coins = CreateCoinsObservable(_balanceChanged)
+			.Publish();
+
+		var coinChanges = coins
+			.ToObservableChangeSet(c => c.HdPubKey.GetHashCode())
+			.AsObservableCache()
 			.Connect()
+			.TransformWithInlineUpdate(x => new WalletCoinViewModel(x))
+			.Publish();
+
+		_isAnySelected = coinChanges
 			.AutoRefresh(x => x.IsSelected)
 			.ToCollection()
-			.Select(items => items.Any(t => t.IsSelected));
+			.Select(items => items.Any(t => t.IsSelected))
+			.ObserveOn(RxApp.MainThreadScheduler)
+			.ToProperty(this, model => model.IsAnySelected)
+			.DisposeWith(disposables);
+		this.RaisePropertyChanged(nameof(IsAnySelected));
```

Comment:
I am not sure, why is this needed?

Reply:
It was a leftover from a previous change. Thanks for noticing :)